llvm/test/CodeGen/AMDGPU/andorbitset.ll

```diff
@@ -4,12 +4,12 @@
 define amdgpu_kernel void @s_clear_msb(ptr addrspace(1) %out, i32 %in) {
 ; SI-LABEL: s_clear_msb:
 ; SI:       ; %bb.0:
-; SI-NEXT:    s_load_dword s4, s[2:3], 0xb
-; SI-NEXT:    s_load_dwordx2 s[0:1], s[2:3], 0x9
+; SI-NEXT:    s_load_dword s6, s[4:5], 0xb
+; SI-NEXT:    s_load_dwordx2 s[0:1], s[4:5], 0x9
 ; SI-NEXT:    s_mov_b32 s3, 0xf000
 ; SI-NEXT:    s_mov_b32 s2, -1
 ; SI-NEXT:    s_waitcnt lgkmcnt(0)
-; SI-NEXT:    s_bitset0_b32 s4, 31
+; SI-NEXT:    s_and_b32 s4, s6, 0x7fffffff
```

Comment:
@arsenm This looks like a big regression. Not sure why that happens.

Reply:
It's only a code size regression. The register allocation changed so now the result isn't the same as the input register. We could add a regalloc hint somewhere to prefer keeping source = dest

Reply:
Okay, then I guess we can just leave as it is for now.